snakePipes/shared/rscripts/sleuth_allelic.R

```diff
@@ -0,0 +1,123 @@
+                   allele = rep(c("genome1", "genome2"), nrow(sample_info)),
+                   condition = rep(sample_info$condition, each = 2) )
+coldata_allelic$allele<-factor(coldata_allelic$allele,levels=c("genome1","genome2"))
+coldata_allelic$condition<-factor(coldata_allelic$condition,levels=unique(coldata_allelic$condition))
+sample_info<-coldata_allelic
+
+if(length(unique(sample_info$condition))>1){
+message("Fitting full model with allele, condition and allele:condition.")
+    d<-formula(~allele + condition + allele:condition)
+#    d0<-formula(~allele + condition)
+}else{
+message("Fitting full model with allele only.")
+    d<-formula(~allele)
+#    d0<-formula(~1)
+}
+colnames(sample_info)[colnames(sample_info) %in% "name"] ="sample"
+print(sample_info)
+sample_info$sample
+
+sample_id = list.dirs(file.path(indir), recursive=F, full.names=F)
+sample_id = sort(sample_id[grep('[^benchmark][^SalmonIndex]', sample_id, invert=F)])
+print(sample_id)
+#sample_id = intersect(sample_info$sample, sample_id) # get only those sample that are defined in the sampleInfo!
+sample_id<-sample_id[match(sample_info$sample,sample_id)]
+print(sample_id)
+
+salmon_dirs = sapply(sample_id, function(id) file.path(indir, id))
+print(salmon_dirs)
+
+s2c = mutate(sample_info, path=salmon_dirs)
+## reorder conditions (for Wald test later on: order of comparison important for fold change)
+s2c$condition<-factor(s2c$condition)
+if ( s2c$condition[[1]] != levels(s2c$condition)[[1]] ) {
+  s2c$condition =  relevel(s2c$condition, as.character(s2c$condition[[1]]) )
+}
+print(s2c)
+
+## get gene names / symbol names
+tryCatch( { t2g = read.table(t2g_file, header=F) },error = function(e) { print('No t2g file available!') },finally = {})
+
+if (exists('t2g')) {
+  colnames(t2g) <- c("target_id","ens_gene","ext_gene")
+
+  ## add gene names
+  so <- sleuth_prep(s2c, full_model=d, target_mapping = t2g, num_cores=6,
+                  transformation_function = function(x) log2(x + 0.5))
+} else {
+  ## construct sleuth object
+  so = sleuth_prep(s2c, full_model=d, num_cores=6,
+                  transformation_function = function(x) log2(x + 0.5))
+}
+
+## model expression responding on condition
+so = sleuth_fit(so)
+#so <- sleuth_fit(so, d, 'full')
+#so <- sleuth_fit(so, d0, 'reduced')
+
+
+# ## fit reduced model (not depending on any factor)
+# so <- sleuth_fit(so, ~1, 'reduced')
+#
+# ## likelihood ratio test (LRT) between models to get transcripts affected by condition.
+# ## Usually considered more stringent than Wald test. No fold change!
+# so <- sleuth_lrt(so, 'reduced', 'full')
+# results_table_lrt <- sleuth_results(so, 'reduced:full', test_type = 'lrt')
+# head(results_table_lrt)
+
+## Wald test (to get *fold change*)
+if(length(unique(sample_info$condition))>1){
+    wald_beta_name = paste0("allelegenome2:condition",unique(coldata_allelic$condition)[2])
+}else{
+    wald_beta_name = "allelegenome2"
+}
```

Comment:
Is it correct you test for the interaction coef when there is condition present (and not allelic change), and for allelic change  alone if there is no condition ?

Reply:
Yes, this would be the desired setup for the allelic mRNAseq. When no condition present, it would only test for the difference between the alleles. This is sort of a technical comparison. 
When condition present, we test for the change in allelic ratio due to the condition level change. This is equivalent to what allelic DESeq2 is doing.